Source/Scene/Scene.js

```diff
@@ -533,6 +535,12 @@ define([
          */
         this.fog = new Fog();
 
+        /**
+         * Render shadows in the scene.
+         * @type {ShadowMap}
+         */
+        this.shadowMap = new ShadowMap(this);
```

Comment:
Passing `this` here feels wrong.  The first thing I thought was that `Scene` and `ShadowMap` were going to be badly tightly coupled.  Perhaps `new ShadowMap(context, new Camerea(this))` is better.